docs/user_manual/components.md

```diff
@@ -18,7 +18,9 @@ The base type for all power-grid-model components.
 
 | name | data type | unit | description                                                                                                                                                       | required |                                     update                                     |
 | ---- | --------- | ---- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------- | :------: | :----------------------------------------------------------------------------: |
-| `id` | `int32_t` | -    | ID of a component. The ID should be unique across all components within the same scenario, e.g., you cannot have a node with `id=5` and another line with `id=5`. | &#10004; | &#10060; (id needs to be specified in the update query, but cannot be changed) |
+| `id` | `int32_t` | -    | ID of a component. The ID should be unique across all components within the same scenario, e.g., you cannot have a node with `id=5` and another line with `id=5`. | &#10004; | &#10060; (If a component update is uniform (See below), the ID can be omitted or set to `nan`. Otherwise, the id needs to be specified in the update query, but cannot be changed) |
+
+A component update is uniform when the number of elements is consistent across all scenarios and matches the number of elements in the input data for that component. Therefore, the ID is not optional for updates on sparse data, and when the ID of a component is ommited, the updated field must be explicit across all scenarios (even if it is not always updated). An example of the usage of optional IDs is given in [Power Flow Example](./Power%20Flow%20Example.ipynb)
```

Comment:
> is uniform when the number of elements is consistent across all scenarios and matches the number of elements in the input data

Not sure if this **was** the case. Updates can all be updating to the same components but not all the components. 

> Therefore, the ID is not optional for updates on sparse data, and when the ID of a component is ommited

Being `uniform` and being `sparse` are two orthogonal concepts I believe. @mgovers could you confirm this?


> the updated field must be explicit across all scenarios (even if it is not always updated).

I do not quite understand this line

Reply:
Oh, it's not yet in review. My apologies.

Reply:
> Being uniform and being sparse are two orthogonal concepts I believe. @mgovers could you confirm this?

see my post in teams in which i elaborate the existing conventions used in the docs 😬 if we can make clear boundaries on terminology, we may resolve this more straightforwardly

Reply:
short version: there's a difference between the API reference and the implementation in main model and we need to agree on one set of conventions

Reply:
> > is uniform when the number of elements is consistent across all scenarios and matches the number of elements in the input data
> 
> Not sure if this **was** the case. Updates can all be updating to the same components but not all the components.

Maybe my explanation wasn't clear, but this intended to reflect the fact that even if you are updating one line (out of 3 for instance), you still need to provide the whole column of the updated attribute in the scenario where it is updated. My intention was to "demonstrate" this in the example notebook.

> > Therefore, the ID is not optional for updates on sparse data, and when the ID of a component is ommited
> 
> Being `uniform` and being `sparse` are two orthogonal concepts I believe. @mgovers could you confirm this?

I see this now, I was following the documentation rather than the code logic. I will update.

> > the updated field must be explicit across all scenarios (even if it is not always updated).
> 
> I do not quite understand this line

I had a hard time trying to write this down concisely and clearly, but it looks like I failed. The line tries to explain what I replied on the first line above.

> short version: there's a difference between the API reference and the implementation in main model and we need to agree on one set of conventions

We can discuss tomorrow, but I believe we should go with the code here.

Please do give me suggestions if you feel I could have written something better, specially regarding the uniform part.

Reply:
I have adjusted in [5bc7cf5](https://github.com/PowerGridModel/power-grid-model/pull/814/commits/5bc7cf5b7b6a24ef5b5540d85703f61db432e830) according to the team discussions about the definitions.